<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="start_button">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>200</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>开始游戏</string>
    </property>
   </widget>
   <widget class="QLabel" name="reversi">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>230</y>
      <width>401</width>
      <height>161</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled/>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>72</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Reversi~</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_1">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>0</y>
      <width>521</width>
      <height>41</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled/>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>   a    b     c    d     e     f     g     h</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>21</width>
      <height>391</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="WindowText">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Text">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled/>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>1

2

3

4

5

6

7

8</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="settings">
    <property name="geometry">
     <rect>
      <x>383</x>
      <y>101</y>
      <width>120</width>
      <height>69</height>
     </rect>
    </property>
    <property name="title">
     <string>设置</string>
    </property>
    <widget class="QMenu" name="menulevel">
     <property name="title">
      <string>level</string>
     </property>
     <addaction name="separator"/>
     <addaction name="level1"/>
     <addaction name="level2"/>
     <addaction name="level3"/>
    </widget>
    <addaction name="menulevel"/>
   </widget>
   <widget class="QMenu" name="display">
    <property name="title">
     <string>显示</string>
    </property>
    <addaction name="actionpossible_move"/>
   </widget>
   <widget class="QMenu" name="save">
    <property name="title">
     <string>存档</string>
    </property>
    <addaction name="actionsave_to_disk"/>
    <addaction name="actioncopy_to_clipboard"/>
   </widget>
   <widget class="QMenu" name="load">
    <property name="title">
     <string>读档</string>
    </property>
    <addaction name="actionimport_from_clipboard"/>
    <addaction name="actionimport_from_disk"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>对局</string>
    </property>
    <addaction name="actionnewgame_player_black"/>
    <addaction name="actionnewgame_player_white"/>
    <addaction name="actionnewgame_ai_play"/>
    <addaction name="actionnewgame_two_player"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="settings"/>
   <addaction name="display"/>
   <addaction name="save"/>
   <addaction name="load"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="level1">
   <property name="text">
    <string>1</string>
   </property>
  </action>
  <action name="level2">
   <property name="text">
    <string>2</string>
   </property>
  </action>
  <action name="level3">
   <property name="text">
    <string>3</string>
   </property>
  </action>
  <action name="actionpossible_move">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>show possible move</string>
   </property>
  </action>
  <action name="actionsave_to_disk">
   <property name="text">
    <string>save to disk</string>
   </property>
  </action>
  <action name="actioncopy_to_clipboard">
   <property name="text">
    <string>copy to clipboard</string>
   </property>
  </action>
  <action name="actionimport_from_clipboard">
   <property name="text">
    <string>load from clipboard</string>
   </property>
  </action>
  <action name="actionimport_from_disk">
   <property name="text">
    <string>load from disk</string>
   </property>
  </action>
  <action name="actionnewgame_player_black">
   <property name="text">
    <string>newgame: player black</string>
   </property>
  </action>
  <action name="actionnewgame_player_white">
   <property name="text">
    <string>newgame: player white</string>
   </property>
  </action>
  <action name="actionnewgame_ai_play">
   <property name="text">
    <string>newgame: ai play</string>
   </property>
  </action>
  <action name="actionnewgame_two_player">
   <property name="text">
    <string>newgame: two player</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>level1</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setLevel1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>level2</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setLevel2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>level3</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setLevel3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionpossible_move</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>enable_poss(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionnewgame_two_player</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionnewgame_ai_play</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>start_game()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionnewgame_player_black</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode1_black()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionnewgame_player_white</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setMode1_white()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionsave_to_disk</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save_game()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionimport_from_disk</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>load_game()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actioncopy_to_clipboard</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save_to_clip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionimport_from_clipboard</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>load_from_clip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setLevel1()</slot>
  <slot>setLevel2()</slot>
  <slot>setLevel3()</slot>
  <slot>enable_poss(bool)</slot>
  <slot>setMode3()</slot>
  <slot>setMode2()</slot>
  <slot>start_game()</slot>
  <slot>setMode1_black()</slot>
  <slot>setMode1_white()</slot>
  <slot>save_game()</slot>
  <slot>load_game()</slot>
  <slot>save_to_clip()</slot>
  <slot>load_from_clip()</slot>
 </slots>
</ui>
